### PR TITLE
ARTEMIS-1365 Advisory consumers listed in Console

### PR DESCRIPTION
**Pull request by gaohoward**
Openwire clients create consumers to advisory topics to receive
notifications. As a result there are internal queues created
on advisory topics. Those consumer shouldn't be exposed via
management APIs which are used by the Console

To fix that the broker doesn't register any queues from
advisory addresses.

### CODE REVIEW

**Comment by andytaylor** [comment]
Howard, This just turns off advisories completely from a management pov, we should have a switch to turn them off or on as well

**Comment by andytaylor** [comment]
Also you need to make sure that the JMX objects aren't created at all for both addresses and queues

**Comment by michaelandrepearce** [comment]
As a user if a remote client attaches to advisory or management addresses we would like to still see those so if something goes rogue we know what is attached.

**Comment by michaelandrepearce** [comment]
Like wise if for some silly reason they attach with a durable we would need to be able to monitor queue depths (jmx)

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/AddressInfo.java [hunk 1]
```diff
@@ -130,6 +130,10 @@ public String toString() {
    }
 
    public boolean isInternal() {
-      return this.name.startsWith(ADVISORY_TOPIC);
+      return isInternal(this.name);
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/AddressInfo.java]: This should be a boolean field internal, that is set by the protocol layer, we shouldn't have protocol specific bits in the core models.

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/AddressInfo.java [hunk 2]
```diff
@@ -130,6 +130,10 @@ public String toString() {
...
+   }
+
+   public static boolean isInternal(SimpleString address) {
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/AddressInfo.java]: Shouldn't be needed should be an attribute on the AddressInfo

**Comment by gaohoward** [comment]
@andytaylor @michaelandrepearce guys agreed please don't merge, I'm making more changes now.
@michaelandrepearce I'm not sure about your query can you rephrase?
Advisories is openwire notification feature. We don't support the full feature, but we do have some implemented (as a by-product required for some other features in amq5 I don't remember exactly thou).

**Comment by michaelandrepearce** [comment]
@gaohoward ill take the question part to dev mail probabky should have done it in first place. And just concentrate on pr details here

**Comment by gaohoward** [comment]
Guys, so I added a parameter to OpenwireProtocolManager, named supportAdvisory. If set to false, no advisory addresses/queues will be created. Default is true.
However I'm not sure about the Console. When supportAdvisory is true, I would still want to hide the JMX objects (not register them but stored them in a list) so that the Console user won't be bothered by them. I think there should be a way to let the user to view those objects, may be adding one operation to the management api? I'm not sure, what you guys think of it?

**Comment by andytaylor** [comment]
As long as the JMX objects aren't created they won't appear on the console,
I would have a flag to disable them completely and one to hide them by not
creating JMX objects

**Comment by gaohoward** [comment]
ok, make sense to me.

**Comment by michaelandrepearce** [comment]
Still would like the logic to sit within openwire protocol not within core classes, addressinfo should just hold a Boolean  flag of if it should be treated as internal address or not.

**Comment by clebertsuconic** [comment]
@andytaylor / @mtaylor  this is an area of your expertise.. I will let you guys merge this one.

**Comment by gaohoward** [comment]
@michaelandrepearce I'll find a way to remove the protocol specific from addressInfo.

**Comment by gaohoward** [comment]
I'll update this again. (Adding some doc about the newly added properties)

**Comment by gaohoward** [comment]
@mtaylor @michaelandrepearce guys I think it's ready for review now. Thanks!

**Comment by andytaylor** [comment]
do we need to move the AddressInfo class?

**Comment by gaohoward** [comment]
@andytaylor got a simple way to keep the AddressInfo. will upload soon after local test check.

**Comment by gaohoward** [comment]
@andytaylor I've made changes to keep AddressInfo in its original package. can you take a look at it?
Thanks.

**Comment by andytaylor** [comment]
looks ok to me

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/SimpleAddressManager.java [hunk 3]
```diff
@@ -316,6 +316,7 @@ private void validateRoutingTypes(SimpleString addressName, Collection<RoutingTy
             if (binding instanceof QueueBinding) {
                final QueueBinding queueBinding = (QueueBinding) binding;
                final RoutingType routingType = queueBinding.getQueue().getRoutingType();
+               System.out.println("got a queue binding........... " + queueBinding.getQueue().getName() + " rt: " + routingType);
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/SimpleAddressManager.java]: Should be removed

File: artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java [hunk 4]
```diff
@@ -1221,6 +1240,7 @@ public Response processWireFormat(WireFormatInfo command) throws Exception {
       public Response processAddDestination(DestinationInfo dest) throws Exception {
          Response resp = null;
          try {
+            System.out.println("___adding dest: " + dest);
```
Comment (michaelandrepearce) [inline artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java]: Should be removed

**Comment by andytaylor** [comment]
@michaelandrepearce it is configured in the broker.xml file via the acceptor url so Im fine with that, dont think we need to persist it. Im happy to merge this.

**Comment by michaelandrepearce** [comment]
@andytaylor can we remove the sysouts before merging.

**Comment by andytaylor** [comment]
@michaelandrepearce yes, I will wait before merging

**Comment by gaohoward** [comment]
Sorry about the System.out, I've remove them.
Thanks guys!

**Comment by michaelandrepearce** [comment]
@gaohoward thanks for all the changes, appreciate the effort. +1 from me.

File: tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/management/OpenWireManagementTest.java [hunk 5]
```diff
@@ -55,6 +87,19 @@ public void setUp() throws Exception {
...
+         //don't set parameters explicitly
+         return;
+      }
+      Set<TransportConfiguration> acceptorConfigs = serverConfig.getAcceptorConfigurations();
+      for (TransportConfiguration tconfig : acceptorConfigs) {
+         if ("netty".equals(tconfig.getName())) {
+            Map<String, Object> params = tconfig.getExtraParams();
+            params.put("supportAdvisory", supportAdvisory);
+            params.put("suppressInternalManagementObjects", suppressJmx);
+            System.out.println("Now use properties: " + params);
```
Comment (michaelandrepearce) [inline tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/management/OpenWireManagementTest.java]: @gaohoward just spotted another

Comment (michaelandrepearce) [reply tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/openwire/management/OpenWireManagementTest.java]: (sorry to be a pain)

**Comment by gaohoward** [comment]
Thanks @michaelandrepearce, great help!
